### PR TITLE
support fixed ntk rope in modeling_rope_utils.py

### PR DESCRIPTION
**Pull request by Fazziekey**
# What does this PR do?

Related PR:
https://github.com/huggingface/transformers/pull/32135
https://github.com/huggingface/transformers/pull/31999

the [Origin NTK scaling factor](https://www.reddit.com/r/LocalLLaMA/comments/14lz7j5/ntkaware_scaled_rope_allows_llama_models_to_have/) is fixed, but now modeling_rope_utils.py only supports  [dynamic ntk](https://www.reddit.com/r/LocalLLaMA/comments/14mrgpr/dynamically_scaled_rope_further_increases/) scaling.

Refer to
Colab code: https://colab.research.google.com/drive/1VI2nhlyKvd5cw4-zHvAIk00cAVj2lCCC#scrollTo=b80b3f37

```
import transformers

old_init = transformers.models.llama.modeling_llama.LlamaRotaryEmbedding.__init__
def ntk_scaled_init(self, dim, max_position_embeddings=2048, base=10000, device=None):

    #The method is just these three lines
    max_position_embeddings = 16384
    a = 8 #Alpha value
    base = base * a ** (dim / (dim-2)) #Base change formula

    old_init(self, dim, max_position_embeddings, base, device)
```

some examples of opensource project:
- [github.com/jquesnelle/yarn](https://github.com/jquesnelle/yarn/blob/995db5b575e75230b3384d658f8b944c9662f775/scaled_rope/LlamaPartNTKScaledRotaryEmbedding.py#L20)
- [github.com/turboderp/exllama](https://github.com/turboderp/exllama/blob/master/model.py#L126)
## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by Fazziekey** [comment]
@Fazziekey

File: src/transformers/modeling_rope_utils.py [hunk 1]
```diff
@@ -158,6 +158,56 @@ def _compute_dynamic_ntk_parameters(
...
+) -> Tuple["torch.Tensor", float]:
+    """
+    Computes the inverse frequencies with NTK scaling. Credits to the Reddit users /u/bloc97 and /u/emozilla
+    Args:
+        config ([`~transformers.PretrainedConfig`]):
+            The model configuration.
+        device (`torch.device`):
+            The device to use for initialization of the inverse frequencies.
+        seq_len (`int`, *optional*):
+            The current sequence length, used to update the dynamic RoPE at inference time.
```
Comment (gante) [inline src/transformers/modeling_rope_utils.py:175]:
```suggestion
            The current sequence length. Unused for this type of RoPE.
```

File: src/transformers/modeling_rope_utils.py [hunk 2]
```diff
@@ -158,6 +158,56 @@ def _compute_dynamic_ntk_parameters(
...
+        base = config.rope_theta
+        partial_rotary_factor = config.partial_rotary_factor if hasattr(config, "partial_rotary_factor") else 1.0
+        dim = int((config.hidden_size // config.num_attention_heads) * partial_rotary_factor)
+        max_position_embeddings = config.max_position_embeddings
+        factor = config.rope_scaling["factor"]
+
+    attention_factor = 1.0  # Unused in this type of RoPE
+
+    # seq_len: default to max_position_embeddings, e.g. at init time
+    seq_len = seq_len if seq_len is not None and seq_len > max_position_embeddings else max_position_embeddings
```
Comment (gante) [inline src/transformers/modeling_rope_utils.py:203]:
```suggestion
```
(unused)

**Review by gante — COMMENTED** [review]
Hey there 👋  Thank you for being proactive and opening a PR! Now that we've abstracted the `inv_freq` computations, we are indeed open to adding more RoPE types!

The PR LGTM. Only one item missing: a numerical test for the new function. See e.g. [this test](https://github.com/huggingface/transformers/blob/b6401030def6e47a64ff99dc073024ddf5ad4707/tests/utils/test_modeling_rope_utils.py#L169) :)

**Comment by Fazziekey** [comment]
> Hey there 👋 Thank you for being proactive and opening a PR! Now that we've abstracted the `inv_freq` computations, we are indeed open to adding more RoPE types!
> 
> The PR LGTM. Only one item missing: a numerical test for the new function. See e.g. [this test](https://github.com/huggingface/transformers/blob/b6401030def6e47a64ff99dc073024ddf5ad4707/tests/utils/test_modeling_rope_utils.py#L169) :)

Hello, I have update test and remove `seq_len`, can you make a review , thanks.

**Comment by Fazziekey** [comment]
@gante can you make a review？ Thanks so much

**Pull request closed without merging**